docker/archive/multi-reader.go

```diff
@@ -0,0 +1,77 @@
+package archive
+
+import (
+	"context"
+
+	"github.com/containers/image/v5/docker/tarfile"
+	"github.com/containers/image/v5/types"
+)
+
+// MultiImageArchiveReader allows for reading a docker-archive with multiple
+// images.  It can be used in subsequent copy.Image operations when set as the
+// copy.Options.ImageSource.  Note that you need to call Next() before each
+// copy.Image operation.
+//
+// For using a MultiImageArchiveReader in copy.Image, set it as the
+// copy.Option.ImageSource.
+type MultiImageArchiveReader struct {
+	*archiveImageSource
+
+	manifests     []tarfile.ManifestItem
+	manifestIndex int
+}
+
+// NewMultiImageArchiveReader returns a new MultiImageArchiveReader based on the reference.
+func NewMultiImageArchiveReader(ctx context.Context, sys *types.SystemContext, ref types.ImageReference) (*MultiImageArchiveReader, error) {
+	path, destinationRef, err := parsePathAndNamedRef(ref.StringWithinTransport())
```

Comment:
This does not really make sense; the user could supply a `dockerReference` here, and it’s going to be accepted?!

`destinationRef` is completely ignored on the read path anyway, right now, and all the user _wanted_ to supply is a file path.